src/Kiota.Builder/Writers/RelativeImportManager.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder.Writers {
+    public class RelativeImportManager {
+        private readonly int prefixLength;
+        private readonly char separator;
+        public RelativeImportManager(string namespacePrefix, char namespaceSeparator)
+        {
+            if(string.IsNullOrEmpty(namespacePrefix))
+                throw new ArgumentNullException(nameof(namespacePrefix));
+            if(namespaceSeparator == default)
+                throw new ArgumentNullException(nameof(namespaceSeparator));
+
+            prefixLength = namespacePrefix.Length;
+            separator = namespaceSeparator;
+        }
+        /// <summary>
+        /// Returns the relative import path for the given using and import context namespace.
+        /// </summary>
+        /// <param name="codeUsing">The using to import into the current namespace context</param>
+        /// <param name="currentNamespace">The current namespace</param>
+        /// <returns>The import symbol, it's alias if any and the relative import path</returns>
+        public (string, string, string) GetRelativeImportPathForUsing(CodeUsing codeUsing, CodeNamespace currentNamespace) {
+            if(codeUsing?.IsExternal ?? true)
+                return (string.Empty, string.Empty, string.Empty);//it's an external import, add nothing
+            var typeDef = codeUsing.Declaration.TypeDefinition;
+
+            var importSymbol = codeUsing.Declaration?.Name?.ToFirstCharacterUpperCase() ?? codeUsing.Name;
+
+            if(typeDef == null)
+                return (importSymbol, codeUsing.Alias, "./"); // it's relative to the folder, with no declaration (default failsafe)
+            else {
+                var importPath = GetImportRelativePathFromNamespaces(currentNamespace, 
+                                                        typeDef.GetImmediateParentOfType<CodeNamespace>());
+                if(string.IsNullOrEmpty(importPath))
+                    importPath+= codeUsing.Name;
+                else
+                    importPath+= codeUsing.Declaration.Name.ToFirstCharacterLowerCase();
+                return (importSymbol, codeUsing.Alias, importPath);
+            }
+        }
+        private string GetImportRelativePathFromNamespaces(CodeNamespace currentNamespace, CodeNamespace importNamespace) {
```

Comment:
All the branches result lead to a return clause. We could probably get rid of the `else` keywords.

Reply:
arrow head vs terminals. I kind of prefer that (more verbose) syntax as it allows me to scan only the if condition and the else block. I don't have to go into the if bock to undeerstand it's content doesn't impact the else block.
If you remove the else, you have to parse the content of the if block. Granted in this case it's fairly easy to do and doesn't add a ton of mental workload.